### ISSUE TITLE
Resize map to fit markers

### ISSUE THREAD

**Issue by spinapse** [bradcornford/Googlmapper#50]
Thanks for the great work on Googlmapper. How would I tell Mapper to set the zoom and boundaries of the map so that it shows all of the markers? I usually do this by calling something like "resize", but I can't figure out the call with Mapper. Or maybe there's a setting when you create the map in the controller?

Thanks.

**Comment by bradcornford** [comment]
Hi, the setting "'center' => false" should move the map to the centre of all markers. The map itself should already update its boundaries based on the attached markers.

```php

Mapper::map(52.381128999999990000, -1.370085000000040000, ['zoom' => 15, 'center' => false, 'makrer' => false']);
Mapper::marker(53.381128999999990000, -1.470085000000040000);
Mapper::marker(53.481128999999990000, -1.570085000000040000);

```

If this doesn't work for you, you could attach "'eventAfterLoad' => '...'" to create your own after load event to resize.

```php

Mapper::map(52.381128999999990000, -1.370085000000040000, ['zoom' => 15, 'center' => false, 'makrer' => false', 'eventAfterLoad' => 'console.log(maps[0].map);']);
Mapper::marker(53.381128999999990000, -1.470085000000040000);
Mapper::marker(53.481128999999990000, -1.570085000000040000);

```

**Comment by spinapse** [comment]
Thanks for the tip. It wasn't working for me, so I added some alerts to slow down the process, and I could see the map correctly, then something after the done call was resetting the bounds incorrectly.

I couldn't track down what was resetting the bounds, so I added a workaround using a variation on your "eventAfterLoad" call. It's shown below, in case it helps anyone else.

The "event" part of the new Map call in the controller:

'eventAfterLoad' => 'delayThenResizeMap(maps[0].map);'

In javascript in the view:

`   var mapInstance;
    var mapBounds;
    var mapZoom;

    function delayThenResizeMap(pMapInstance)
    {
        mapInstance = pMapInstance;
        mapBounds = mapInstance.getBounds();
        mapZoom = mapInstance.getZoom();
        setTimeout(resizeMap, 100);
    }

    function resizeMap()
    {
        mapInstance.fitBounds(mapBounds);
        mapInstance.setZoom(mapZoom);
    }
`
